R/Model-class.R

```diff
@@ -1230,227 +1231,142 @@ DualEndpoint <- function(mean,
 
-                       ## the intercept (= first location betaW value)
-                       betaWintercept ~ dnorm(0, 0.000001)
-                       ## ~ essentially dflat(),
-                       ## which is not available in JAGS.
-                   })
-
-    if(useRW1)
-    {
-        ## add RW1 part
-        start@priormodel <-
-            h_jags_join_models(start@priormodel,
-                       function(){
-                           ## the iid first oder differences:
-                           for (j in 2:nGrid) {
-                               delta[j-1] ~ dnorm(0, precBetaW / (doseGrid[j] - doseGrid[j-1]))
-                           }
-                       })
-    } else {
-        ## add RW2 part
-        start@priormodel <-
-            h_jags_join_models(start@priormodel,
-                       function(){
-                           ## the first order differences:
-                           delta[1] <- deltaStart
-                           for (j in 2:(nGrid-1)) {
-                               delta[j] <- deltaStart +
-                                   sum(delta2[1:(j-1)])
-                           }
-
-                           ## the iid second oder differences:
-                           for (j in 1:(nGrid-2)) {
-                               delta2[j] ~ dnorm(0, 2 * precBetaW / (doseGrid[j+2] - doseGrid[j]))
-                           }
-
-                           ## the first 1st order difference:
-                           deltaStart ~ dnorm(0, 0.000001)
-                       })
-    }
-
-    ## we will fill in more, depending on which parameters
-    ## are fixed, in these two variables:
-    start@sample <- c(start@sample,
-                      "betaWintercept", "betaW", "delta")
-
-    ## copy some things that we need to amend
-    oldModelspecs <- start@modelspecs
-    oldInit <- start@init
-
-    ## check the variance for the RW prior
-    if(! start@use_fixed["sigma2betaW"])
-    {
-        start@priormodel <-
-            h_jags_join_models(start@priormodel,
-                       function(){
-                           ## gamma prior for RW precision
-                           precBetaW ~ dgamma(precBetaWa, precBetaWb)
-                       })
-
-        start@sample <- c(start@sample,
-                          "precBetaW")
-
-        start@init <-
-            function(y, w, nGrid){
-                c(oldInit(y, w, nGrid),
-                  list(precBetaW=1))
-            }
-
-        start@modelspecs <-
-            function(){
-                c(oldModelspecs(),
-                  list(precBetaWa=sigma2betaW["a"],
-                       precBetaWb=sigma2betaW["b"]))
-            }
-
-    } else {
-        start@modelspecs <-
-            function(){
-                c(oldModelspecs(),
-                  list(precBetaW=1/sigma2betaW))
-            }
-    }
+  start@priormodel <- if (rw1) {
+    h_jags_join_models(
```

Comment:
since priormodel is being joined in both if and else blocks, how about first just assigning the rw part (with if / else) and then doing the join?

Reply:
Sure, great idea, Tx!